Source/ToolCore/Build/BuildLinux.cpp

```diff
@@ -0,0 +1,182 @@
+// THE SOFTWARE.
+//
+
+#include <Atomic/Core/StringUtils.h>
+#include <Atomic/IO/Log.h>
+#include <Atomic/IO/FileSystem.h>
+#include <Atomic/Resource/ResourceCache.h>
+
+#include "../ToolSystem.h"
+#include "../ToolEnvironment.h"
+#include "../Project/Project.h"
+#include "../Project/ProjectSettings.h"
+#include "../Assets/AssetDatabase.h"
+
+#include "BuildLinux.h"
+#include "BuildSystem.h"
+#include "BuildEvents.h"
+
+namespace ToolCore
+{
+
+BuildLinux::BuildLinux(Context * context, Project *project) : BuildBase(context, project, PLATFORMID_LINUX)
+{
+
+}
+
+BuildLinux::~BuildLinux()
+{
+
+}
+
+void BuildLinux::Initialize()
+{
+    ToolSystem* tsystem = GetSubsystem<ToolSystem>();
+    Project* project = tsystem->GetProject();
+
+    Vector<String> defaultResourcePaths;
+    GetDefaultResourcePaths(defaultResourcePaths);
+    
+    for (unsigned i = 0; i < defaultResourcePaths.Size(); i++)
+    {
+        AddResourceDir(defaultResourcePaths[i]);
+    }
+    BuildDefaultResourceEntries();
+    
+    // Include the project resources and cache separately
+    AddProjectResourceDir(project->GetResourcePath());
+    AssetDatabase* db = GetSubsystem<AssetDatabase>();
+    String cachePath = db->GetCachePath();
+    AddProjectResourceDir(cachePath);
+
+    BuildProjectResourceEntries();
+}
+
+bool BuildLinux::CheckIncludeResourceFile(const String& resourceDir, const String& fileName)
+{
+    // #623 BEGIN TODO: Skip files that have a converted version in the cache
+    AssetDatabase* db = GetSubsystem<AssetDatabase>();
+    String cachePath = db->GetCachePath();
+    if (resourceDir != cachePath)
+    {
+        String ext = GetExtension(fileName);
+        if (ext == ".jpg" || ext == ".png" || ext == ".tga")
+        {
+            FileSystem* fileSystem = GetSubsystem<FileSystem>();
+            String compressedPath = cachePath + "DDS/" + fileName + ".dds";
+            if (fileSystem->FileExists(compressedPath))
+                return false;
+        }
+    }
+    // #623 END TODO
+    
+    return BuildBase::CheckIncludeResourceFile(resourceDir, fileName);
+}
+
+void BuildLinux::BuildNative(const String& buildPath)
+{
+    BuildLog("Building Linux Application");
+
+    ToolEnvironment* tenv = GetSubsystem<ToolEnvironment>();
+
+    String playerBinary = tenv->GetPlayerBinary();
+
+    if (!BuildCopyFile(playerBinary, buildPath_ + "/AtomicPlayer"))
+        return;
+        
+// the BuildCopyFile does not copy a file from one location to another,
+// instead it opens a new file, and copies the contents. This results
+// in the loss of file execution bits/acls. There is no cross platform
+// way to transferr or restore these on the copied file.
+#ifdef ATOMIC_PLATFORM_LINUX 
+    // linux-to-linux restore the execute permissions lost in copy
+    FileSystem* fileSystem = GetSubsystem<FileSystem>();
+    if (fileSystem->SystemCommand ( "chmod +x " + buildPath_ + "/AtomicPlayer") ) 
+        return;
+#endif
```

Comment:
Would a chmod call be possible here instead of shelling out to system chmod command? http://man7.org/linux/man-pages/man2/chmod.2.html, I like the idea of explicitly setting the permission, even if we were doing a file copy, permissions can still get stripped pretty easily by packaginging, etc